rest/src/rest/RESTEmitter.cpp

```diff
@@ -262,12 +262,18 @@ auto zpt::RESTEmitter::has_pending(zpt::json _envelope) -> bool {
 auto zpt::RESTEmitter::reply(zpt::json _request, zpt::json _reply) -> void {
 	auto _exists = this->__pending.find(std::string(_request["channel"]));
 	if (_exists != this->__pending.end()) {
+		auto _callback = _exists->second;
+		this->__pending.erase(_exists);
 		if (_reply->ok()) {
-			auto _callback = _exists->second;
-			this->__pending.erase(_exists);
-			_callback(zpt::ev::Reply, std::string(_request["resource"]), _reply, this->self());
+			try {
+				_callback(zpt::ev::Reply, std::string(_request["resource"]), _reply, this->self());
+			} catch(zpt::assertion& _e) {
+				_callback(zpt::ev::Reply, std::string(_request["resource"]), { "status", _e.status(), "payload", { "text", _e.what(), "assertion_failed", _e.description(), "code", _e.code() } }, this->self());
+			} catch(std::exception& _e) {
+				_callback(zpt::ev::Reply, std::string(_request["resource"]), { "status", 500, "payload", { "text", _e.what(), "code", 0 } }, this->self());
+			}
```

Comment:
Must be missing something, sorry about that. Really need to understand why we're notifying the API's client callback of an exception that occurred in that same callback. This logic seems a bit fuzzy since it would stand to reason that the exception should be dealt with within the callback itself, if needed be.

Reply:
There is no real reason in a perfect scenario, it is just for protection/better understanding where is failing and not let the request without a response in case of failure and close the connection as soon as possible (because the callback is not controlled by Zapata, it can be a code with some error that will cause stuck requests, because it will not be replied and the socket will be waiting for a response), as it doesn't add any complexity and help in the debug process when something goes wrong I think that will not be a big problem. But you can change it if it doesn't please you :)

Reply:
Thank you for the explanation! 
Let's not do it, then, we're re-invoking a function that we know for a fact thrown an exception. If the API's client needs to process the exception, that can be done in the callback itself.
Thank you!